### PR TITLE
Vagrant box build fix

### PR DESCRIPTION
**Pull request by tj90241**
`packer` can leave artifacts in `libvirt` that prevent box
builds from succeeding in some cases, without leaving an
appropriate error message.

Perform some pre-flight activities to increase the success
rate of box builds.

### CODE REVIEW

File: virtual/packer/bin/create-packer-box.sh [hunk 1]
```diff
@@ -37,6 +37,10 @@ base_box_exists=$(
     || true
 )
 if [ -z "$base_box_exists" ]; then
+    # prevent vagrant-libvirt from failing if there's scrapnel lying around
```
Comment (jingqgeng) [inline virtual/packer/bin/create-packer-box.sh:40]: Should we check if the base box provider is libvirt before we call virsh commands?

Comment (justinjpacheco) [reply virtual/packer/bin/create-packer-box.sh:40]: @jingqgeng i believe the `|| true` will always succeed regardless of virsh being there or not.

Comment (justinjpacheco) [reply virtual/packer/bin/create-packer-box.sh:40]: never mind, i just tried this and it errored out

Comment (justinjpacheco) [reply virtual/packer/bin/create-packer-box.sh:40]: never mind again. while it does show an error message, the error code is still 0

```
> which virsh
> virsh || true
-bash: virsh: command not found
> echo $?
0
> virsh
-bash: virsh: command not found
> echo $?
127
```

**Review by jingqgeng — DISMISSED** [review]

**Comment by tj90241** [comment]
Had to also add `virsh destroy ...`, otherwise noticed that it would undefine domains but leave them running in some cases.

**Comment by tj90241** [comment]
That's a very good idea, thanks, I'll add that.

**Comment by tj90241** [comment]
Sorry for all the pushes, think I covered all the above.

Also discovered that the volume name that we attempt to purge seems off, perhaps some new vagrant/packer change.

**Review by comay — DISMISSED** [review]

**Review by jingqgeng — APPROVED** [review]

**Review by comay — APPROVED** [review]

**Pull request merged**